File: leet_code/math/504/504.py
class Solution:
    def convertToBase7(self, num: int) -> str:
        if num == 0:
            return "0"
        
        negative = num < 0
        n = abs(num)
        digits = []
        
        while n > 0:
            digits.append(str(n % 7))
            n //= 7
        
        result = ''.join(digits[::-1])
        return '-' + result if negative else result

File: leet_code/math/504/504.dart
class Solution {
  String convertToBase7(int num) {
    if (num == 0) return "0";

    bool negative = num < 0;
    int n = num.abs();
    StringBuffer sb = StringBuffer();

    while (n > 0) {
      sb.write(n % 7);
      n ~/= 7;
    }

    String result = sb.toString().split('').reversed.join('');
    return negative ? '-$result' : result;
  }
}
